src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -8,77 +8,70 @@ namespace Microsoft.AspNetCore.Razor.TagHelpers
-        public TagHelperAttribute(IReadOnlyTagHelperAttribute attribute)
-            : this(attribute?.Name, attribute?.Value)
+        /// <param name="name">The <see cref="Name"/> of the attribute.</param>
+        /// <param name="value">The <see cref="Value"/> of the attribute.</param>
+        public TagHelperAttribute(string name, object value)
+            : this(name, value, minimized: false)
         {
-            if (attribute == null)
-            {
-                throw new ArgumentNullException(nameof(attribute));
-            }
-
-            Minimized = attribute.Minimized;
         }
 
         /// <summary>
-        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>
-        /// and <paramref name="value"/>.
+        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>,
+        /// <paramref name="value"/> and <paramref name="minimized"/>.
         /// </summary>
         /// <param name="name">The <see cref="Name"/> of the attribute.</param>
         /// <param name="value">The <see cref="Value"/> of the attribute.</param>
-        public TagHelperAttribute(string name, object value)
+        /// <param name="minimized">The <see cref="Minimized"/> value of the attribute.</param>
+        public TagHelperAttribute(string name, object value, bool minimized)
         {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
             Name = name;
             Value = value;
+            Minimized = minimized;
         }
 
         /// <summary>
         /// Gets or sets the name of the attribute.
         /// </summary>
-        public string Name { get; set; }
+        public string Name { get; }
 
         /// <summary>
         /// Gets or sets the value of the attribute.
         /// </summary>
-        public object Value { get; set; }
+        public object Value { get; }
 
         /// <summary>
         /// Gets or sets an indication whether the attribute is minimized or not.
         /// </summary>
         /// <remarks>If <c>true</c>, <see cref="Value"/> will be ignored.</remarks>
-        public bool Minimized { get; set; }
+        public bool Minimized { get; }
 
-        /// <summary>
-        /// Converts the specified <paramref name="value"/> into a <see cref="TagHelperAttribute"/>.
-        /// </summary>
-        /// <param name="value">The <see cref="Value"/> of the created <see cref="TagHelperAttribute"/>.</param>
-        /// <remarks>Created <see cref="TagHelperAttribute"/>s <see cref="Name"/> is set to <c>null</c>.</remarks>
-        public static implicit operator TagHelperAttribute(string value)
-        {
-            return new TagHelperAttribute
-            {
-                Value = value
-            };
-        }
```

Comment:
I agree we can't have this `operator` with the other changes you're making. But has the public API change been agreed to?